lib/src/analyzers/lint_analyzer/rules/flutter_rule_utils.dart

```diff
@@ -0,0 +1,49 @@
+import 'package:analyzer/dart/element/type.dart';
+import 'package:collection/collection.dart';
+
+bool hasWidgetType(DartType type) =>
+    isWidgetOrSubclass(type) ||
+    _isIterable(type) ||
+    _isList(type) ||
+    _isFuture(type);
+
+bool isWidgetOrSubclass(DartType? type) =>
+    _isWidget(type) || _isSubclassOfWidget(type);
+
+bool isStateOrSubclass(DartType? type) =>
```

Comment:
@incendial please rename to isWidgetStateOrSubclass

Reply:
Fixed.